src/main/objects/RulesChecker.js

```diff
@@ -208,8 +208,26 @@ function checkRules(prevPrevChord, prevChord, currentChord, rules, checkSumJumpR
     return result
 }
 
+function checkDelayCorrectness(prevChord, currentChord){
+    var delay = prevChord.harmonicFunction.delay;
+    for(var i=0; i<delay.length; i++){
+        var prevComponent = delay[i][0];
+        var currentComponent = delay[i][1];
+        var voice = -1;
+        for(var i=0; i<4; i++){
+            if(prevChord.notes[i].chordComponent === prevComponent) {
+                if(currentChord.notes[i].chordComponent !== currentComponent) return -1;
+            }
+            else{
+                if(!prevChord.notes[i].equals(currentChord.notes[i] || i === 0)) return -1;
```

Comment:
ten or ma na pewno być w środku?

Reply:
Tak, bo bas może się zmienić (np. zmienić oktawę), a pozostałe głosy muszą pozostać na tych samych wysokościach

Reply:
To w prevChord.notes[i] jest boolean?

Reply:
I porównujemy go z currentChord.notes[i] lub i == 0 ?

Reply:
A okej, chodzi Ci o brak nawiasu.

Reply:
Tak, chodziło mi o to, że jest or w equals
Mogłem lepiej to napisać...